modules/vpc/src/test/java/com/ibm/cloud/is/vpc/v1/model/DedicatedHostPrototypeDedicatedHostByZoneTest.java

```diff
@@ -80,7 +81,7 @@ public void testDedicatedHostPrototypeDedicatedHostByZone() throws Throwable {
     assertEquals(dedicatedHostPrototypeDedicatedHostByZoneModelNew.name(), "my-host");
     assertEquals(dedicatedHostPrototypeDedicatedHostByZoneModelNew.profile().toString(), dedicatedHostProfileIdentityModel.toString());
     assertEquals(dedicatedHostPrototypeDedicatedHostByZoneModelNew.resourceGroup().toString(), resourceGroupIdentityModel.toString());
-    assertEquals(dedicatedHostPrototypeDedicatedHostByZoneModelNew.group().toString(), dedicatedHostGroupPrototypeDedicatedHostByZoneContextModel.toString());
+    assertEquals(JsonParser.parseString(dedicatedHostPrototypeDedicatedHostByZoneModelNew.group().toString()), JsonParser.parseString(dedicatedHostGroupPrototypeDedicatedHostByZoneContextModel.toString()));
```

Comment:
We are comparing two strings(.toString()), both extending same `GenericModel` class.

Is this needed ? When I ran the tests, it ran fine.

Reply:
You are right that currently it won't cause an issue. However, the method toString() will implicitly call `HashMap`'s iterator. On Java's documentation, it said

> Hash table based implementation of the Map interface. This implementation provides all of the optional map operations, and permits null values and the null key. (The HashMap class is roughly equivalent to Hashtable, except that it is unsynchronized and permits nulls.) ***This class makes no guarantees as to the order of the map; in particular, it does not guarantee that the order will remain constant over time.***

The behavior of that method is thus non-deterministic over time, so if some fields are in a different order in the string, the comparison will fail without having any real problems.

The change is just a preventive one, in case false alarms arise in future Java versions. It's done with relatively small changes to add in an extra bit of robusticity without affecting the original intentions of the test.